src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/H2Engine.java

```diff
@@ -385,10 +385,16 @@ protected MappedEntity createPreparedStatementForInserts(final DbEntity entity)
         final List<String> values = new ArrayList<>();
         final List<String> columnsWithAutoInc = new ArrayList<>();
         final List<String> valuesWithAutoInc = new ArrayList<>();
+        String columnWithAutoIncName = null;
         for (final DbColumn column : entity.getColumns()) {
             columnsWithAutoInc.add(quotize(column.getName()));
             valuesWithAutoInc.add("?");
-            if (!column.isAutoInc()) {
+            if (column.isAutoInc()) {
+                if (columnWithAutoIncName == null) {
```

Comment:
you can join these conditions or even delete the second one because it's confirmed that a DBEntity can only have 1 AutoInc Column